node/src/block_producer/block_producer_reducer.rs

```diff
@@ -545,7 +557,9 @@ impl BlockProducerEnabled {
             &global_slot_since_genesis,
         );
         let consensus_state = v2::ConsensusProofOfStakeDataConsensusStateValueStableV2 {
-            blockchain_length: v2::UnsignedExtendedUInt32StableV1((pred_block.height() + 1).into()),
+            blockchain_length: v2::UnsignedExtendedUInt32StableV1(
+                (pred_block.height().saturating_add(1)).into(),
```

Comment:
@dkuehr I am thinking if cases like these shouldn't be fatal failures. If this happens the state will be messed up and we cannot trust it, continuing as if nothing happened will just delay the error until later making things more confusing.